### DISCUSSION TITLE
[@types/react-table] I think UseTableRowProps is wrongly defined

### DISCUSSION THREAD

**Discussion by danielo515** [DefinitelyTyped/DefinitelyTyped#60688] [Issues with a @types package] [answered]
This issue is about react-table, and the type definition of UseTableRowProps.
Currently, this is the definition:

```typescript
export interface UseTableRowProps<D extends object> {
    cells: Array<Cell<D>>;
    allCells: Array<Cell<D>>;
    values: Record<IdType<D>, CellValue>;
    getRowProps: (propGetter?: RowPropGetter<D>) => TableRowProps;
    index: number;
    original: D;
    id: string;
    subRows: Array<Row<D>>;
}
```
The problem, IMO, lies in the typing of the original property.
[According to the library docs](https://react-table-v7.tanstack.com/docs/api/useTable#row-properties), the original property is the specific entry in the data array that originates that row.
So I think the proper definition would be as follows:

```typescript
export interface UseTableRowProps<D extends object> {
    cells: Array<Cell<D>>;
    allCells: Array<Cell<D>>;
    values: Record<IdType<D>, CellValue>;
    getRowProps: (propGetter?: RowPropGetter<D>) => TableRowProps;
    index: number;
    original: D.data[number];
    id: string;
    subRows: Array<Row<D>>;
}


```

[a comment by an automated account is not shown]

**Comment by ggascoigne** [comment, marked as answer]
I don't think so.

The input data is defined as `D[]`, an array of some type `D`.  `row.original` is "The original row object from the `data` array that was used to materialize this row."  Based on the docs, and based on dumping out the row data, the correct type for the `original` is type `D`.

**Reply by danielo515** [reply]
Yes, you're right.
I have mistakenly been using this types as they were the props passed to the table component. Because of that they were not making much sense. Typescript error messages didn't helped either, and they confused me to the point where I didn't know what I was looking at.

Sorry
